inference-engine/tests/functional/inference_engine/ir_serialization/const_compression.cpp

```diff
@@ -0,0 +1,263 @@
+    std::ifstream bin_1(m_out_bin_path_1, std::ios::in | std::ios::binary);
+
+    int shape_element_count = 1;
+    std::for_each(shape.begin(), shape.end(), [&](const int &n) {shape_element_count *= n;});
+    ASSERT_TRUE(file_size(bin_1) == unique_const_count * shape_element_count * sizeof(int));
+}
+
+TEST_F(SerializatioConstantCompressionTest, NonIdenticalConstants) {
+    std::shared_ptr<ngraph::Function> ngraph_a;
+    const int unique_const_count = 2;
+    ngraph::Shape shape{2, 2, 2};
+
+    auto A = ngraph::op::Constant::create(ngraph::element::i32, shape,
+        {1, 2, 3, 4, 5, 6, 7, 8});
+    auto B = ngraph::op::Constant::create(ngraph::element::i32, shape,
+        {2, 2, 3, 4, 5, 6, 7, 8});
+
+    ngraph_a = std::make_shared<ngraph::Function>(ngraph::NodeVector{A, B},
+        ngraph::ParameterVector{});
+
+    ngraph::pass::Serialize(m_out_xml_path_1, m_out_bin_path_1).run_on_function(ngraph_a);
+
+    std::ifstream xml_1(m_out_xml_path_1, std::ios::in | std::ios::binary);
+    std::ifstream bin_1(m_out_bin_path_1, std::ios::in | std::ios::binary);
+
+    int shape_element_count = 1;
+    std::for_each(shape.begin(), shape.end(), [&](const int &n) {shape_element_count *= n;});
+    ASSERT_TRUE(file_size(bin_1) == unique_const_count * shape_element_count * sizeof(int));
+}
+
+TEST_F(SerializatioConstantCompressionTest, IdenticalConstantsDifferentTypesI32I64) {
+    std::shared_ptr<ngraph::Function> ngraph_a;
+    const int unique_const_count = 1;
+    ngraph::Shape shape{2, 2, 2};
+
+    auto A = ngraph::op::Constant::create(ngraph::element::i32, shape,
+        {1, 0, 2, 0, 3, 0, 4, 0});
+    auto B = ngraph::op::Constant::create(ngraph::element::i64, ngraph::Shape({1, 2, 2}),
+        {1, 2, 3, 4});
+
+    ngraph_a = std::make_shared<ngraph::Function>(ngraph::NodeVector{A, B},
+        ngraph::ParameterVector{});
+
+    ngraph::pass::Serialize(m_out_xml_path_1, m_out_bin_path_1).run_on_function(ngraph_a);
+
+    std::ifstream xml_1(m_out_xml_path_1, std::ios::in | std::ios::binary);
+    std::ifstream bin_1(m_out_bin_path_1, std::ios::in | std::ios::binary);
+
+    int shape_element_count = 1;
+    std::for_each(shape.begin(), shape.end(), [&](const int &n) {shape_element_count *= n;});
+    ASSERT_TRUE(file_size(bin_1) == unique_const_count * shape_element_count * sizeof(int));
+}
+
+TEST_F(SerializatioConstantCompressionTest, IdenticalConstantsDifferentTypesI32I8) {
+    std::shared_ptr<ngraph::Function> ngraph_a;
+    const int unique_const_count = 1;
+
+    ngraph::Shape shape{2, 2, 2};
+    auto A = ngraph::op::Constant::create(ngraph::element::i32, shape,
+        {1, 2, 3, 4, 5, 6, 7, 8});
+    auto B = ngraph::op::Constant::create(ngraph::element::i8, ngraph::Shape({2, 4, 4}),
+        {1, 0, 0, 0,
+         2, 0, 0, 0,
+         3, 0, 0, 0,
+         4, 0, 0, 0,
+         5, 0, 0, 0,
+         6, 0, 0, 0,
+         7, 0, 0, 0,
+         8, 0, 0, 0});
```

Comment:
For more complicated shape I will suggest to switch to `ngraph::test::NDArray`

Reply:
So, I made the shape less complicated